src/index.test.ts

```diff
@@ -3060,6 +3060,488 @@ describe('TypedDataUtils.sanitizeData', function () {
   });
 });
 
+describe('TypedDataUtils.eip712Hash', function () {
+  describe('V3', function () {
+    it('should hash a minimal valid typed message', function () {
+      const hash = sigUtil.TypedDataUtils.eip712Hash(
+        {
+          types: {
+            EIP712Domain: [],
+          },
+          primaryType: 'EIP712Domain',
+          domain: {},
+          message: {},
+        },
+        'V3',
+      );
+
+      expect(hash.toString('hex')).toMatchSnapshot();
+    });
+
+    it('minimal typed message hash should be identical to minimal valid typed message hash', function () {
+      const minimalHash = sigUtil.TypedDataUtils.eip712Hash(
+        {
+          types: {},
+          primaryType: 'EIP712Domain',
+        } as any,
+        'V3',
+      );
+      const minimalValidHash = sigUtil.TypedDataUtils.eip712Hash(
+        {
+          types: {
+            EIP712Domain: [],
+          },
+          primaryType: 'EIP712Domain',
+          domain: {},
+          message: {},
+        },
+        'V3',
+      );
+
+      expect(minimalHash.toString('hex')).toBe(
+        minimalValidHash.toString('hex'),
+      );
+    });
+
+    it('should ignore extra top-level properties', function () {
+      const minimalValidHash = sigUtil.TypedDataUtils.eip712Hash(
+        {
+          types: {
+            EIP712Domain: [],
+          },
+          primaryType: 'EIP712Domain',
+          domain: {},
+          message: {},
+        },
+        'V3',
+      );
+      const extraPropertiesHash = sigUtil.TypedDataUtils.eip712Hash(
+        {
+          types: {
+            EIP712Domain: [],
+          },
+          primaryType: 'EIP712Domain',
+          domain: {},
+          message: {},
+          extra: 'stuff',
+          moreExtra: 1,
+        } as any,
+        'V3',
+      );
+
+      expect(minimalValidHash.toString('hex')).toBe(
+        extraPropertiesHash.toString('hex'),
+      );
+    });
+
+    it('should hash a typed message with a domain separator that uses all fields.', function () {
```

Comment:
Nit: most test descriptions don't have a period
```suggestion
    it('should hash a typed message with a domain separator that uses all fields', function () {
```